### PR TITLE
fix(mcp): Auto-register .envrc files for secret management

### PR DESCRIPTION
**Pull request by aaronsteers**
# fix(mcp): Auto-register .envrc files for secret management

## Summary
Fixes an issue where MCP tools were prompting for credentials manually instead of automatically loading them from `.envrc` files. The problem was that PyAirbyte's default secret managers only looked for `.env` files, not `.envrc` files.

The fix adds automatic detection and registration of `.envrc` files in the MCP `initialize_secrets()` function. When a `.envrc` file exists in the current working directory, it gets registered as a `DotenvSecretManager` and loaded into the environment.

**Before**: MCP tools prompted for `AIRBYTE_CLOUD_WORKSPACE_ID`, `AIRBYTE_CLOUD_CLIENT_ID`, and `AIRBYTE_CLOUD_CLIENT_SECRET` even when they were defined in `.envrc`

**After**: MCP tools automatically load credentials from `.envrc` without prompting

## Review & Testing Checklist for Human
- [ ] **Test with different working directories** - Verify `.envrc` detection works when MCP tools are invoked from different directories
- [ ] **Test edge cases** - Verify behavior with missing `.envrc`, malformed `.envrc`, or `.envrc` with syntax errors
- [ ] **Verify secret manager priority** - Confirm that `.envrc` secrets have the intended precedence over other secret sources
- [ ] **Security review** - Consider security implications of automatically loading `.envrc` files from the current directory
- [ ] **Test broader MCP functionality** - Verify other MCP tools still work correctly (I only tested `check_airbyte_cloud_workspace` and `list_deployed_cloud_connections`)

### Test Plan
1. Run `poe mcp-tool-test check_airbyte_cloud_workspace '{}'` - should work without prompting
2. Run `poe mcp-tool-test list_deployed_cloud_connections '{}'` - should work without prompting  
3. Test from different directories to verify path resolution
4. Test with other MCP tools to ensure no regressions

### Notes
- This change affects all MCP operations since it modifies the core secret initialization
- The fix registers the `.envrc` secret manager after custom dotenv managers but before GCP GSM
- Both secret registration AND environment loading occur for `.envrc` files

---
**Link to Devin run**: https://app.devin.ai/sessions/3dbb42ea20ff42e59cb1a635753c2407    
**Requested by**: @aaronsteers

### CODE REVIEW

File: airbyte/_util/meta.py [hunk 1]
```diff
@@ -28,10 +31,33 @@ def get_colab_release_version() -> str | None:
...
+def is_mcp_mode() -> bool:
+    """Return True if running in MCP (Model Context Protocol) mode."""
+    if _MCP_MODE_ENABLED:
+        return True
+
+    script_name = get_python_script_name()
+    if script_name and "airbyte-mcp" in script_name:
+        return True
+
+    return "fastmcp" in sys.modules
```
Comment (aaronsteers) [inline airbyte/_util/meta.py]: aside -

```suggestion
    # Else, we are not running in an MCP context.
    return False
```

File: airbyte/_util/meta.py [hunk 2]
```diff
@@ -28,10 +31,30 @@ def get_colab_release_version() -> str | None:
     return None
 
 
+@lru_cache
```
Comment (aaronsteers) [inline airbyte/_util/meta.py]: aside - 


```suggestion
```

File: tests/unit_tests/test_progress.py [hunk 3]
```diff
@@ -161,6 +161,9 @@ def test_ci_environment_a_progress_style(monkeypatch):
 def test_ci_environment_b_progress_style(monkeypatch):
     """Test the style in a CI environment."""
     monkeypatch.setenv("CI", "1")
+    from airbyte._util import meta
+
+    meta.is_ci.cache_clear()
```
Comment (aaronsteers) [inline tests/unit_tests/test_progress.py]: aside - 

```suggestion
```

File: airbyte/_util/meta.py [hunk 4]
```diff
@@ -32,6 +35,25 @@ def is_ci() -> bool:
...
+    _MCP_MODE_ENABLED = True
+
+
+def is_mcp_mode() -> bool:
+    """Return True if running in MCP (Model Context Protocol) mode."""
+    if _MCP_MODE_ENABLED:
+        return True
+
+    script_name = get_python_script_name()
+    return bool(script_name and "airbyte-mcp" in script_name)
```
Comment (aaronsteers) [inline airbyte/_util/meta.py]: aside - 

```suggestion
    return _MCP_MODE_ENABLED
```
